router/core/graphql_handler.go

```diff
@@ -195,6 +199,13 @@ func (h *GraphQLHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			w.Header().Set("Cache-Control", "no-store, no-cache, must-revalidate")
 		}
 
+		if h.exprVisitorManager.IsResponseBodyUsedInExpressions() {
+			// String returns the unread portion of the buffer
+			// However at this point the offset is 0, which means
+			// we will get the entire string
+			reqCtx.expressionContext.Response.Body.Raw = respBuf.String()
```

Comment:
Are you sure that this is the right place and method? We want to log the final response. This would not catch errors in the pre_handler or when something manipulates the response body via custom module

Reply:
Good point, will check

Reply:
New proposal
```
ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor) // existing code

// NEW TEE READER
if responseBodyUsedInExpression {
  var respBuf bytes.Buffer
  ww.Tee(&respBuf)
}

h.handler.ServeHTTP(ww, r) // existing code

// set to response body
if responseBodyUsedInExpression {
  expr.Response.Body = respBuf.String()
}
```

Since we write to bytes.Buffer should be fast enough, but will benchmark to be sure, let me know what you think

Reply:
It internally calls tee.Write https://github.com/go-chi/chi/blob/master/middleware/wrap_writer.go#L96 discard is never true, as we won't call Discard() on the wrap writer.